### ISSUE TITLE
A minor incompatibility with very old Flask versions has crept in

### ISSUE THREAD

**Issue by RookieRick** [plangrid/flask-rebar#74]
In working to address #73 (in which some tests were broken because Travis CI used latest version of Werkzeug, which tweaked error messages and - more importantly - changed from 301 to 308 for redirect on missing trailing slash), I have stumbled across another issue.

Adjusting the trailing slash test to accept either 301 or 308, I now get Travis build errors specifically for Flask 0.10 (but not for 0.11+).  It would seem that Flask 0.10 does not handle the 308 correctly, specifically, it fails to maintain the same content-type.  All of which leads to a few questions that I thought might make sense to discuss here.

Biggest question: Does anyone see a need to fully support Flask 0.10 (nearly 6 years old)?  I'm hard-pressed to imagine a real-world scenario in which somebody would be using Flask 0.10 and trying to use something as modern and shiny as Flask-Rebar. ;-)

Probably the simplest way for me to unblock on this issue is just to remove that from Travis CI tests.  So, we wouldn't technically "require" 0.11+ but it would make it on par with pre-0.10 versions..  Not technically "required" but also not tested as part of our standard build.

Does that cause anyone heartburn? (cc @barakalon - sorry for Flask-Rebar spam today LOL, but I'm definitely interested in your opinion as the father of this library.  Any additional insights into how particular pairings of Flask and Marshmallow were selected when building .travis.yml would be much appreciated!)

**Comment by barakalon** [comment]
First, a note on why https://github.com/plangrid/flask-rebar/pull/73 might've failed:
It might have something to do with this: https://github.com/plangrid/flask-rebar/blob/master/flask_rebar/rebar.py#L59
We might just need to tweak that a bit to be 301 _or_ 308.

Second, regarding the bigger question: **Do we need to support Flask 0.10?**

Here's what we know:
- Maintaining support for Flask 0.10 is a pain
- Library support is part of our packages contract to consumers
- Removing library support is a breaking change
- Breaking changes require bumping the major version of Flask-Rebar

We could sneak the removal of this support without bumping the major version if we were confident that no one is using Flask 0.10. However, since this is open source, there's no good way to know that.

So, I think that leaves us with two options:
(1) Deal with the pain for now and make the change work with Flask 0.10
(2) Remove support for Flask 0.10 and bump the major version of Flask-Rebar

My gut says maintaining support hasn't gotten painful enough yet and we should go with option (1).

Maybe it will get painful enough, and we can remove it. Or maybe we'll have another reason to release a major version (Open API 3.0?), at which point we could remove support.

**Comment by RookieRick** [comment]
Thanks, Barak - appreciate the perspective, and I hadn't noticed that bit you pointed to.  Will check that out!

**Comment by RookieRick** [comment]
So, yeah.. you nailed it.  The issue was actually potentially a bit more than just "breaking our unit tests" so I've adjusted that pre-0.11 bit to now treat 308 same as 301 and all seems right with the world again.  Thanks again!
